pkg/reconciler/cache/replication/replication_reconcile.go

```diff
@@ -84,8 +89,39 @@ func (c *controller) reconcile(ctx context.Context, gvrKey string) error {
 			func(gvr schema.GroupVersionResource, cluster logicalcluster.Name, namespace, name string) (interface{}, error) {
 				return retrieveCacheObject(&gvr, c.globalWorkspaceTypeIndexer, c.shardName, cluster, namespace, name)
 			},
-			func(cluster logicalcluster.Name, _, name string) (interface{}, error) {
-				return c.localWorkspaceTypeLister.Cluster(cluster).Get(name)
+			func(cluster logicalcluster.Name, _, name string) (interface{}, bool, error) {
+				obj, err := c.localWorkspaceTypeLister.Cluster(cluster).Get(name)
+				return obj, true, err
+			})
+	case rbacv1.SchemeGroupVersion.WithResource("clusterroles").String():
+		return c.reconcileObject(ctx,
+			keyParts[1],
+			rbacv1.SchemeGroupVersion.WithResource("clusterroles"),
+			rbacv1.SchemeGroupVersion.WithKind("ClusterRole"),
+			func(gvr schema.GroupVersionResource, cluster logicalcluster.Name, namespace, name string) (interface{}, error) {
+				return retrieveCacheObject(&gvr, c.globalClusterRoleIndexer, c.shardName, cluster, namespace, name)
+			},
+			func(cluster logicalcluster.Name, _, name string) (interface{}, bool, error) {
+				obj, err := c.localClusterRoleLister.Cluster(cluster).Get(name)
+				if err != nil {
+					return obj, true, err
+				}
+				return obj, obj.Annotations[core.ReplicateAnnotationKey] != "", err
+			})
+	case rbacv1.SchemeGroupVersion.WithResource("clusterrolebindings").String():
+		return c.reconcileObject(ctx,
+			keyParts[1],
+			rbacv1.SchemeGroupVersion.WithResource("clusterrolebindings"),
+			rbacv1.SchemeGroupVersion.WithKind("ClusterRoleBinding"),
+			func(gvr schema.GroupVersionResource, cluster logicalcluster.Name, namespace, name string) (interface{}, error) {
+				return retrieveCacheObject(&gvr, c.globalClusterRoleBindingIndexer, c.shardName, cluster, namespace, name)
+			},
+			func(cluster logicalcluster.Name, _, name string) (interface{}, bool, error) {
+				obj, err := c.localClusterRoleBindingLister.Cluster(cluster).Get(name)
+				if err != nil {
+					return obj, true, err
```

Comment:
I think this line causes the panic, replace with `return nil, false, err`

Reply:
Isn't obj nil then?

Reply:
done